SmartDeviceLink/private/SDLProtocolReceivedMessageProcessor.m

```diff
@@ -0,0 +1,317 @@
+//
+//  SDLProtocolProcessMessageByte.m
+//  SmartDeviceLink
+//
+//  Created by George Miller on 7/13/22.
+//  Copyright © 2022 smartdevicelink. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import "SDLProtocolHeader.h"
+#import "SDLProtocolReceivedMessageProcessor.h"
+#import "SDLProtocolReceivedMessageRouter.h"
+#import "SDLSecurityType.h"
+
+@implementation SDLProtocolReceivedMessageProcessor
+
+-(id)init {
+    self = [super init];
+    version = 0;
+    encrypted = false;
+    frameType = 0;
+    dataLength = 0;
+    dataBytesRemaining = 0; //Counter for the data pump
+    
+
+    // Message management
+    endOfMessage = 0;
+    header = nil;
+
+    //Reset state
+    [self ResetState];
+    return self;
+}
+
+- (void)ResetState{
+    // Flush Buffers
+    self.headerBuffer = [NSMutableData dataWithCapacity:0];
+    self.payloadBuffer = [NSMutableData dataWithCapacity:0];
+    dataBytesRemaining = 0;
+    
+    // Reset state
+    state = START_STATE;
+    prevState = ERROR_STATE;
+
+}
+
+// Loop through the given bytes and call the state machien to process each byte.
+- (void)stateMachineManager:(NSData *)receiveBuffer withBlock:(stateMachineMessageReadyBlock)messageReadyBlock{
+    //get a pointer to the bytes because NSMutableData is layered
+    const char *bytes = [receiveBuffer bytes];
+    
+    //now loop through the bytes
+    for (int i = 0; i < [receiveBuffer length]; i++)
+    {
+        //hand the byte to the state machine
+        endOfMessage = [self sdl_processMessagesStateMachine:(Byte)bytes[i]];
+        // If we have reached the end of a message, we need to imediatly process that completion block, then reset the buffers and keep pumping data
+        if (endOfMessage){
+            // Reset flag
+            endOfMessage = 0;
+            
+            // Pass the header and payload to the completion
+            messageReadyBlock(header.encrypted, header, [NSData dataWithData:self.payloadBuffer]);
+            
+            //Reset state for the next message
+            [self ResetState];
+        }
+    }
+    
+}
+
+// This is the state machine
+// It processes a single byte of a message, checks for errors,
+// and builds up a header buffer and a payload buffer
+// When the header and payload are complete, the message is processed
+// The state of the state machine effectively tracks which byte of a message we are expecting next
+// For reference: https://smartdevicelink.com/en/guides/sdl-overview-guides/protocol-spec/
+// If a byte comes in that does not conform to spec, the buffers are flushed and state is reset.
```

Comment:
Comments should be documentation comments here. You should also generally not cut off comments and let word wrap do it's job. See other comments for how this is handled in this project.